opal/mca/accelerator/rocm/accelerator_rocm_module.c

```diff
@@ -476,6 +478,44 @@ static int mca_accelerator_rocm_get_device(int *dev_id)
     return OPAL_SUCCESS;
 }
 
+static int mca_accelerator_rocm_get_device_pci_attr(int dev_id, opal_accelerator_pci_attr_t *pci_attr)
+{
+    hipError_t err;
+    int ret;
+    static const int PCI_BUS_ID_LENGTH = 13;
+    char pci_bus_id[PCI_BUS_ID_LENGTH];
+    char domain_id[5] = {0}, bus_id[3] = {0}, device_id[3] = {0}, function_id[2] = {0};
+
+    if (NULL == pci_attr) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    err = hipDeviceGetPCIBusId(pci_bus_id, PCI_BUS_ID_LENGTH, dev_id);
+    if(hipSuccess != err) {
+        opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
+                            "error retrieviung device PCI attributes");
+        return OPAL_ERROR;
+    }
+
+    ret = sscanf(pci_bus_id, "%4s:%2s:%2s.%s", domain_id, bus_id, device_id, function_id);
```

Comment:
The last string (function_id) in sscanf doesn't have a length.  Is it variable size?

Reply:
Thank you. Fixed.

It should be 1 char, e.g. `0000:00:02.0`